capstone/src/main/java/com/google/sps/servlets/TagsTFIDFServlet.java

```diff
@@ -144,21 +192,59 @@ public void calculateTFIDF(LinkedHashMap<Long, LinkedHashMap<String, Integer>> g
     }
   }
 
+  /**
+   * Weights the tf-idf score of a given ngram by taking into account its length.
+   */
+  private double weightScore(double score, String ngram) {
+    double wordsWeight = (ngram.split(" ").length / 3.1);
+    double lengthWeight = ((double) ngram.length() / longestNgramLength);
```

Comment:
last thing we discussed was being able to weight these and they them out, so wordsWeight * wordsWeightScalar where scalar is a value [0, 1], and same for stringLengthScalar. You could run a simple eval where you set these values to a handful of different values, and figure out which fits best.

Reply:
Yeah, I tried that by changing the coefficients of the weight terms on this line:
`double totalWeight = 1 + wordsWeight + lengthWeight;`
It didn't really make much of a difference, besides changing up the orders of the top terms in a few cases. In basically every case it didn't cause the top terms to be any different, so I left it at 1. Although now that you point it out, I should probably change the weights to 0.5 for both instead.